org.eclipse.dawnsci.hdf5/src/org/eclipse/dawnsci/hdf5/HDF5Utils.java

```diff
@@ -45,6 +54,43 @@
 public class HDF5Utils {
 	private static final Logger logger = LoggerFactory.getLogger(HDF5Utils.class);
 
+	private static String host;
+
+	/**
+	 * @return local host name
+	 */
+	public static String getLocalHostName() {
+		if (host == null) {
+			ExecutorService ex = Executors.newSingleThreadExecutor();
+			Future<String> fu = ex.submit(new Callable<String>() {
+				@Override
+				public String call() throws Exception {
+					try {
+						return InetAddress.getLocalHost().getHostName();
+					} catch (UnknownHostException e) {
+						logger.error("Could not find host name", e);
+					}
+					return null;
+				}
+			});
+	
+			try {
+				host = fu.get(5, TimeUnit.SECONDS);
+			} catch (TimeoutException e) {
+				logger.error("Timed out finding host name", e);
+			} catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)